src/parser/Tokenizer.h

```diff
@@ -179,5 +179,62 @@ class Tokenizer
 
 } /* namespace Parser */
 
+#include "sbuf/Stream.h"
+
+/// parses a decimal integer that fits the specified Integer type
+template <typename Integer>
+static Integer
+ParseInteger(const char *description, const SBuf &rawInput)
```

Comment:
Please split into Parser::SignedDecimalInteger() and Parser::UnsignedDecimalInteger() (both accepting signed and unsigned integer types but only the first one accepting negative input values) and move both into new src/parser/ToInteger.{cc,h} source files.

The templated common code may go into Parser::DecimalInteger_<Integer>() helper function.

The non-templated common code may go into Parser::DecimalInteger_() helper function which will have int64_t limits as parameters and so it will not clash with the other helper function.

We want to discourage direct use of helper functions by adding that underscore suffix to their names. If there is a C++ trick that allows the compiler to prohibit all DecimalInteger_() use outside of Parser::SignedDecimalInteger() and Parser::UnsignedDecimalInteger() context, then we should use that trick.

Reply:
> Please split into ...

Done. In my implementation SignedDecimalInteger<>() just calls DecimalInteger_<>() - so we can eliminate the second one, I guess.

> If there is a C++ trick that allows the compiler to prohibit all DecimalInteger_() use 

I am not aware of such trick. Can we create a nested namespace (with a name emphasizing its 'internal' scope)  and move this function there?